### ISSUE TITLE
New version of Core engine causes errors in USDM rule execution

### ISSUE THREAD

**Issue by BSnoeijerCD** [cdisc-org/cdisc-rules-engine#1651]
### Standard

CDISC USDM

### Reference Rule ID(s)

DDF00141, DDF00093, DDF00094

### Conformance Rule ID(s) (if published in CORE)

CORE-000857, CORE-000873, CORE-000874

### JIRA Ticket

https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-

### CLI Command Used (if applicable)

python core.py -s SDTM -v 3.4 -d /path/to/datasets

### Rule YAML

```YAML
Authorities: 
  - Organization: 
    Standards:
      - Name: 
        References:
          - Citations:
              - Cited Guidance: 
                Document: 
                Item: 
                Section: 
            Origin: 
            Rule Identifier: 
              Id: 
              Version: 
            Version: 
        Version:
Check:
  all:
    - name: 
      operator: 
      value: 
Core:
  Id: 
  Status: 
  Version: '1'
Description: 
Executability: Fully Executable
Match Datasets:
  - Keys:
      - 
    Name: 
Operations:
  - id: 
    operator: 
Outcome:
  Message: 
  Output Variables:
    -
Rule Type: 
Scope:
Sensitivity:
```

### Attach any sample test data file(s)

Standard Fixed json version gives error for gender + client version where studyVersion.date is missing (which is allowed).

### Attach any output report and/or log file(s)

[Test_CoreMar26_fixedFile.xlsx](https://github.com/user-attachments/files/25799159/Test_CoreMar26_fixedFile.xlsx)

+ client output (cannot be shared) gives "EXECUTION ERROR" for rule DDF00093 and DDF00094

### Expected output

SUCCESS run - no issues.
